### PR TITLE
Add CoreDNS proposals

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds CoreDNS DoQ/DoH3 support as a proposal for sutdent consideration.

### CODE REVIEW

File: summerofcode/2023.md [hunk 1]
```diff
@@ -157,6 +159,18 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
 - Upstream Issue (URL): https://github.com/buildpacks/pack/issues/1623
 
+### CoreDNS
+
+#### Add DNS-over-QUIC (DoQ) and/or DNS-over-HTTP/3 (DoH3) support
+
+- Description: DNS-over-QUIC (DoQ) and DNS-over-HTTP/3 (DoH3) are relatively new protocols for transmitting DNS queries with security and privacy. Additionally, DoQ and DoH3 also offers other benefits such as improved latency and better error detection. The goal of this proposal is to add DoQ and/or DoH3 support to CoreDNS.
+- Expected outcome: An implementation of DoQ or DoH3 for CoreDNS. A stretch goal of adding both DoQ and DoH3 is also within scope.
+- Recommended Skills: Golang, DNS.
+- Mentor(s): Yong Tang (@yongtang)
```
Comment (aliok) [inline summerofcode/2023.md:169]: Could you please add your email?

**Review by aliok — COMMENTED** [review]
It looks good. I confirmed @yongtang is in https://github.com/cncf/foundation/blob/main/project-maintainers.csv#L60

We would just need your email @yongtang to be able to add you to the GSoC system.

**Comment by yongtang** [comment]
Thanks @aliok . I have updated the PR with email added. Let me know if anything else is needed.

**Review by nate-double-u — APPROVED** [review]
Thanks for this @yongtang!

**Pull request merged**